app/src/androidTest/java/ch/epfl/polybazaar/SaleDetailsTest.java

```diff
@@ -62,5 +72,57 @@ public void testFillWithListingAndGetSellerInfo() throws Throwable {
             assertEquals("gu.vrut@epfl.ch", contactSel.getText().toString());
         });
     }
+
+    @Test
+    public void testWithMockListing() {
+        useMockDataStore();
+
+        String listingID1 = randomUUID().toString();
+        String listingID2 = randomUUID().toString();
+
+        Intent intent = new Intent();
+        intent.putExtra("listingID", listingID1);
+
+        activityRule.launchActivity(intent);
+        Listing newListing = new Listing("Title", "description", "0.0", "test.user@epfl.ch", "");
+        LiteListing newLiteListing = new LiteListing(listingID1, newListing.getTitle(), newListing.getPrice(), newListing.getCategory());
+        ListingImage listingImage1 = new ListingImage(convertBitmapToString(convertDrawableToBitmap(ContextCompat.getDrawable(activityRule.getActivity(), R.drawable.ue_roll))), listingID2);
+        ListingImage listingImage2 = new ListingImage(convertBitmapToString(convertDrawableToBitmap(ContextCompat.getDrawable(activityRule.getActivity(), R.drawable.ue_roll))), "");
+
+        storeListing(newListing, listingID1, result -> {
+            assertEquals(true, result);
+            addLiteListing(newLiteListing, resultLite -> {
+                assertEquals(true, resultLite);
+                storeListingImage(listingImage1, listingID1, resultImage1 -> {
+                    assertEquals(true, resultImage1);
+                    storeListingImage(listingImage2, listingID2, resultImage2 -> {
+                        assertEquals(true, resultImage2);
+                        try {
+                            runOnUiThread(() -> {
+                                //recreate to load the new Listing
+                                activityRule.getActivity().recreate();
+                            });
+                        } catch (Throwable throwable) {
+                            throwable.printStackTrace();
+                        }
+                        try {
+                            Thread.sleep(1000);
```

Comment:
Is this necessary for testing the UI? Just asking. It's not for the DB request right?

Reply:
Yes, otherwise the callback isn't done in time and the test failed.